pydantic/functional_validators.py

```diff
@@ -266,28 +329,39 @@ def __call__(
     )
     _V2WrapValidatorType = TypeVar('_V2WrapValidatorType', bound=Union[_V2WrapValidator, _PartialClsOrStaticMethod])
 
+FieldValidatorModes: TypeAlias = Literal['before', 'after', 'wrap', 'plain']
+
 
 @overload
 def field_validator(
     field: str,
     /,
     *fields: str,
-    mode: Literal['before', 'after', 'plain'] = ...,
+    mode: Literal['wrap'],
     check_fields: bool | None = ...,
-) -> Callable[[_V2BeforeAfterOrPlainValidatorType], _V2BeforeAfterOrPlainValidatorType]: ...
+    input_type: Any = ...,
+) -> Callable[[_V2WrapValidatorType], _V2WrapValidatorType]: ...
 
 
 @overload
 def field_validator(
     field: str,
     /,
     *fields: str,
-    mode: Literal['wrap'],
+    mode: Literal['before', 'plain'],
     check_fields: bool | None = ...,
-) -> Callable[[_V2WrapValidatorType], _V2WrapValidatorType]: ...
+    input_type: Any = ...,
+) -> Callable[[_V2BeforeAfterOrPlainValidatorType], _V2BeforeAfterOrPlainValidatorType]: ...
 
 
-FieldValidatorModes: TypeAlias = Literal['before', 'after', 'wrap', 'plain']
+@overload
+def field_validator(
+    field: str,
+    /,
+    *fields: str,
+    mode: Literal['after'] = ...,
+    check_fields: bool | None = ...,
+) -> Callable[[_V2BeforeAfterOrPlainValidatorType], _V2BeforeAfterOrPlainValidatorType]: ...
 
 
 def field_validator(
```

Comment:
Changes to this function look great, nice work with the consistent changes to the overloads.